middleware/transaction.go

```
@@ -0,0 +1,49 @@
+				panic(r) // パニックを再スロー
+			}
+		}()
+
+		c.Next()
+
+		// レスポンスのステータスコードを確認
+		if c.Writer.Status() >= 400 {
+			tx.Rollback()
+		} else {
+			if err := tx.Commit(); err != nil {
+				tx.Rollback()
```

golangci-lint (1.64.8)

Error return value of `tx.Rollback` is not checked

(errcheck)